tests/unit/test_charm.py

```diff
@@ -450,58 +382,38 @@ def test_nginx_replan_with_synapse_service_not_existing(
 ) -> None:
     """
     arrange: start Synapse charm with Synapse container but without synapse service,
-        and mock replan_nginx.
+        and mock restart_nginx.
     act: Fire that NGINX container is ready.
     assert: Pebble Service replan NGINX is called but unit is in maintenance waiting for Synapse.
     """
     harness.begin()
-    replan_nginx_mock = MagicMock()
-    monkeypatch.setattr(pebble, "replan_nginx", replan_nginx_mock)
+    restart_nginx_mock = MagicMock()
+    monkeypatch.setattr(pebble, "restart_nginx", restart_nginx_mock)
 
     harness.container_pebble_ready(synapse.SYNAPSE_NGINX_CONTAINER_NAME)
 
-    replan_nginx_mock.assert_called_once()
+    restart_nginx_mock.assert_called_once()
     assert harness.model.unit.status == ops.MaintenanceStatus("Waiting for Synapse")
 
 
-def test_redis_relation_pebble_success(redis_configured: Harness, monkeypatch: pytest.MonkeyPatch):
+def test_redis_relation_success(redis_configured: Harness, monkeypatch: pytest.MonkeyPatch):
     """
     arrange: start the Synapse charm, set server_name, mock synapse.enable_redis.
     act: emit redis_relation_updated
-    assert: synapse.enable_redis is called once and unit is active.
+    assert: synapse.enable_redis is called once.
     """
     harness = redis_configured
     enable_redis_mock = MagicMock()
     monkeypatch.setattr(synapse, "enable_redis", enable_redis_mock)
-    pull_mock = MagicMock(return_value=io.StringIO("{}"))
-    push_mock = MagicMock()
-    container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
-    monkeypatch.setattr(container, "pull", pull_mock)
-    monkeypatch.setattr(container, "push", push_mock)
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+    container.push(
+        synapse.SYNAPSE_CONFIG_PATH, f'server_name: "{TEST_SERVER_NAME}"', make_dirs=True
+    )
     harness.begin()
 
     harness.charm.on.redis_relation_updated.emit()
 
     enable_redis_mock.assert_called_once()
-    assert isinstance(harness.model.unit.status, ops.ActiveStatus)
-
-
-def test_redis_relation_pebble_error(redis_configured: Harness, monkeypatch: pytest.MonkeyPatch):
```

Comment:
Why isn't this scenario tested anymore?

Reply:
Same as other observers, they are not interacting with pebble.enable_<observer> anymore. Only reconcile.